Makefile

```
@@ -8,6 +8,7 @@ export GO111MODULE := on
 export GOPRIVATE := github.com/mesosphere/kubeaddons,github.com/mesosphere/ksphere-testing-framework
 export KBA_KUBECONFIG ?= $(shell mktemp --tmpdir kba-kubeconfig-XXXXXXXX)
 export KUBECONFIG = $(KBA_KUBECONFIG)
+export KBA_BRANCH ?= $(shell git branch | grep -v detached | awk '$$1=="*"{print $$2}')
```

```suggestion
export KBA_BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
```
or
```suggestion
export KBA_BRANCH ?= $(shell git branch --show-current)
```
The latter only works with [git 2.22+](https://github.com/git/git/blob/master/Documentation/RelNotes/2.22.0.txt#L32).

---

The latter was my first attempt, failed because of 2.22+ so not using it. The former does not work if we are detached, if thats ok then ill use it. I do not think this was worth blocking this PR for though?

---

> failed because of 2.22+

That's unfortunate.

> former does not work if we are detached

I assume we'd never work on a detached HEAD, do we?

>  I do not think this was worth blocking this PR for though?

I'm not sure we have the same understanding of what "request changes" means.

IMO the `awk` solution is much more error prone because it depends on the output formatting and we've had issues with git output being formatted in a way that the original author didn't anticipate (e.g. `git show` including GPG signatures).

---

Maybe we can just update git in the image that runs this job?

---

> Maybe we can just update git in the image that runs this job?

I rather make one change at a time, for now just trying to get this test to work in a sane way. There are some good number of assumptions and issues with the makefile in general and i do not want to make all of the changes now. But yeah this would be worth while.

---

> I'm not sure we have the same understanding of what "request changes" means.

I agree, we do not. Should we discuss in a team meeting for more details about what this means?